src/pages/lesson-plans/step.ts

```diff
@@ -0,0 +1,42 @@
+import {Component, EventEmitter, Input, Output} from "@angular/core";
+
+
+let nextStepId: number = 0;
+let nextQuestionId: number = 0;
+
+@Component({
+  selector: 'step',
+  templateUrl: 'step.html'
+})
+export class Step {
+
+  stepId: number;
+  questionId?: number;
+  done: boolean = false;
+
+  @Input() allSteps: Step[];  // there might be a better way of doing this
+  @Input() questionType?: string;
+  @Input() correctAnswer?: string;  // sometimes, a question will not have a correct answer
+  @Input() highlightedStepId: number;
+  @Input() currentQuestionStepId: number;
+
+  @Output() onGetResponses = new EventEmitter<number>();
+
+
+  constructor() {
+    this.stepId = nextStepId++;
+    if (this.questionType) {
+      // not all steps are questions
+      this.questionId = nextQuestionId++;
+    }
+    this.allSteps[this.stepId] = this;
+  }
+
+  clickStep() {
+    if (!this.questionType) {
+      this.done = !this.done;
+      return;
+    }
+    this.onGetResponses.emit(this.stepId);
```

Comment:
this file is pretty much complete, at least if things don't need to be changed significantly from my vision.  Most of the real effort would take place in LessonPlan or the teacher's classroom session view